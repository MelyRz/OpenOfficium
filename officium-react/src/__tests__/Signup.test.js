import React from 'react';
import { shallow } from '../enzyme';
import { BrowserRouter as Router } from 'react-router-dom';
import { createMount, createShallow } from '@material-ui/core/test-utils'
import Signup from '../components/Signup'

import { GoogleLoginButton } from "react-social-login-buttons";
import { TwitterLoginButton } from "react-social-login-buttons";
import { GithubLoginButton } from "react-social-login-buttons";
import { FacebookLoginButton } from "react-social-login-buttons";

describe('Signup Page Tests', () => {
  let shallow;
  let mount;

  beforeAll(() => {
    mount = createMount();
    shallow = createShallow();
  });

  afterAll(() => {
    mount.cleanUp();
  })

  it ('Signup snapshot', () => {
    const wrapper = shallow(<Signup/>);

    expect(wrapper).toMatchSnapshot();
  })

  it ('Renders signUp button', () => {
    const wrapper = mount(<Router><Signup/></Router>);
    const button = wrapper.findWhere(n => n.type() === 'button' && n.text() === 'SIGN UP');
    expect(button.exists()).toBe(true);
  })


  it ('Renders google Signup button', () => {
    const wrapper = mount(<Router><Signup/></Router>);

    expect(wrapper.find(GoogleLoginButton).exists()).toBe(true);
  })

  it ('Renders twitter Signup button', () => {
    const wrapper = mount(<Router><Signup/></Router>);

    expect(wrapper.find(TwitterLoginButton).exists()).toBe(true);
  })

  it ('Renders github Signup button', () => {
    const wrapper = mount(<Router><Signup/></Router>);

    expect(wrapper.find(GithubLoginButton).exists()).toBe(true);
  })
  it ('Renders facebook Signup button', () => {
    const wrapper = mount(<Router><Signup/></Router>);

    expect(wrapper.find(FacebookLoginButton).exists()).toBe(true);
  })

})
