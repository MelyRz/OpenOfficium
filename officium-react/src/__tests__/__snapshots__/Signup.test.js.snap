// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Signup Page Tests Signup snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <WithStyles(withRouter()) />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "classes": Object {
        "button": "withRouter---button-6",
        "container": "withRouter---container-1",
        "error": "withRouter---error-7",
        "field": "withRouter---field-5",
        "logo": "withRouter---logo-2",
        "paper": "withRouter---paper-4",
        "social": "withRouter---social-3",
      },
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "classes": Object {
          "button": "withRouter---button-6",
          "container": "withRouter---container-1",
          "error": "withRouter---error-7",
          "field": "withRouter---field-5",
          "logo": "withRouter---logo-2",
          "paper": "withRouter---paper-4",
          "social": "withRouter---social-3",
        },
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "context": Object {},
  },
  Symbol(enzyme.__childContext__): null,
}
`;
